File: morpheus_core/config/Networking/Wireless/wifiConnect.py
#!/usr/bin/python
import os
import sys
try:
	ssid=sys.argv[1]
	password=sys.argv[2]
	encryption=sys.argv[3]
except:
	print "Syntax: \n"
	print "./wifiConnect.py <ssid> <password> <wpa/wep>"
	sys.exit()
def connect(ssid,password,encryption):
	if (encryption=="wep"):
		os.system("nmcli dev wifi connect '"+ssid  + "' wep-key-type "+ password)
	elif (encryption=="wpa"):
		os.system("nmcli dev wifi connect '"+ssid + "' password " + password)
	elif (encryption=="--"):
		os.system("nmcli dev wifi connect '"+ssid +  "' iface wlan0")
	else:
		print "\nInvalid encryption type. Please either type wpa or wep"
		sys.exit()

connect(ssid,password,encryption)

File: centurion_core/WebApp/php/wificonnect.php
<?php
$sessid = $_COOKIE["sessionid"];
$sessions = file_get_contents("/etc/auth/sessions.json");
$sess_json = json_decode($sessions,true);
$valid = "false";
foreach($sess_json as $key => $value){
	if($value==$sessid && $sessid!="-1"){
	$valid = "true";
	}
}
if($valid == "true"){
	$ssid = $_POST["ssid"];
	$mode = $_POST["mode"];
	$password = $_POST["password"];

	if($mode == "encryption"){
		$cmd = "sudo nmcli dev wifi list | grep '".$ssid."' | awk '{print \$NF}'";
		exec($cmd,$output);  
		echo $output[0];
	}
	if($mode == "connect"){
		$cmd = "sudo nmcli dev wifi list | grep '".$ssid."' | awk '{print \$NF}'";
		exec($cmd,$output);  
		$encryption=$output[0];
		if($encryption=="WPA2" || $encryption=="WPA1"){
			$connectcmd = "sudo wifi-connect '".$ssid."' '".$password."' wpa";
			exec($connectcmd,$connectoutput);
			echo $connectoutput[0];

		}elseif($encryption=="WEP"){
			$connectcmd = "sudo wifi-connect '".$ssid."' '".$password."' wep";
			exec($connectcmd,$connectoutput);
			echo $connectoutput[0];
		}else{
			echo "Unsupported encryption type";
		}
	}
}
?>
